### ISSUE TITLE
cp.where() failed on complex matrices

### ISSUE THREAD

**Issue by liangs91** [cupy/cupy#2164]
I tried to use cp.where() on complex matrices, which seems not supported yet. Here is a snippet of the code
```
import cupy as cp
a = cp.random.rand(2,2).astype(cp.complex64)
b = cp.random.rand(2,2).astype(cp.complex64)
c = cp.array([[0,0],[1,1]])
d = cp.where(c,a,b)
```
the error message looks like
```
 File "<stdin>", line 1, in <module>
  File "/home/liangs/anaconda3/envs/tensorflow/lib/python3.5/site-packages/cupy/sorting/search.py", line 136, in where
    return _where_ufunc(condition.astype('?'), x, y)
  File "cupy/core/_kernel.pyx", line 833, in cupy.core._kernel.ufunc.__call__
  File "cupy/core/_kernel.pyx", line 704, in cupy.core._kernel._guess_routine
TypeError: Wrong type ((<class 'numpy.bool_'>, <class 'numpy.complex64'>, <class 'numpy.complex64'>)) of arguments for cupy_where
```

**Comment by kmaehashi** [comment]
@liangs91 Thanks for reporting an issue. Do you have any interest in making a PR for this case?

**Comment by AntoineDujardin** [comment]
I had the same issue. Adding `'?FF->F', '?DD->D'` in the `ops` of the `_where_ufunc` in `cupy/sorting/search.py` solved the problem in my case (CUDA 9.2).
I guess I could make a PR, but wouldn't that be overkill?

**Comment by kmaehashi** [comment]
@AntoineDujardin Thanks! It would be great if you can make a PR.
You may also need to remove `no_complex=True` from `where` tests defined in `cupy_tests/sorting_tests/test_search.py`.

**Comment by AntoineDujardin** [comment]
Done

**Comment by asi1024** [comment]
Merged #2175.
